test/vc-data-model-1.0/20-advanced.js

```diff
@@ -36,48 +35,104 @@ describe('Advanced Documents', () => {
+        should.exist(doc.credentialSchema);
+        let schemas = [].concat(doc.credentialSchema);
+        expect(schemas.length > 0).to.be.true;
+
+        for(let schema of schemas) {
+          schema.type.should.be.a('string');
+        }
       });
+
       it.skip('value of `type` MUST be defined in the active context / term dictionary', async () => {
         // test for `type`'s value existence in the active context / term dictionary
       });
-      it.skip('MUST specify an `id` property', async () => {
+
+      it('MUST specify an `id` property', async () => {
         // test for `id` property existence
+        const doc = await util.generate('example-010.jsonld', generatorOptions);
+        should.exist(doc.credentialSchema);
+        let schemas = [].concat(doc.credentialSchema);
+        expect(schemas.length > 0).to.be.true;
+
+        for(let schema of schemas) {
+          schema.id.should.be.a('string');
+        }
       });
-      it.skip('value of `id` MUST be a URI identifying a schema file', async () => {
+
+      it('value of `id` MUST be a URI identifying a schema file', async () => {
         // test that `id`'s value is a valid URI
         // TODO: https://github.com/w3c/vc-data-model/issues/381
+        const doc = await util.generate('example-010.jsonld', generatorOptions);
+        const schemas = [].concat(doc.credentialSchema);
+
+        for(let schema of schemas) {
+          expect(schema.id).to.match(uriRegex);
+        }
       });
     });
   });
 
   // https://w3c.github.io/vc-data-model/#refreshing
   describe('Refreshing', () => {
-    it.skip('`refreshService` MUST provide one or more refresh services', async () => {
+    it('`refreshService` MUST provide one or more refresh services', async () => {
       // test that `refreshService` is either an array or an object
+      const doc = await util.generate('example-011.jsonld', generatorOptions);
+      const isArray = Array.isArray(doc.refreshService) &&
+        doc.refreshService.length > 0;
+      const isObject = doc.refreshService && typeof doc.refreshService.id === 'string';
+      expect(isArray || isObject).to.be.true;
     });
+
     describe('each object within `refreshService`...', () => {
       // if there are multiple objects, loop these tests
-      it.skip('MUST specify a `type` property with a valid value', async () => {
-        // test for `type` property exitence
+      it('MUST specify a `type` property with a valid value', async () => {
+        // test for `type` property existence
+        const doc = await util.generate('example-011.jsonld', generatorOptions);
+        const services = [].concat(doc.refreshService);
+
+        for(let service of services) {
+          expect(service.type).to.be.a('string');
+        }
       });
+
       it.skip('value of `type` MUST be defined in the active context / term dictionary', async () => {
         // test for `type`'s value existence in the active context / term dictionary
```

Comment:
These `type` values need to be checked against the type's expressed in the `@context` files OR they may be URIs: https://w3c.github.io/vc-data-model/#dfn-type

A minimal test would be to first check that the `type` is a URI--if it is, pass the test. If the `type` is not a URI, then check that it's value(s) exist as keys in any of the references `@context` objects.

The `@context` look-up bit could be added to `utils` probably as this will be needed elsewhere.